app/components/works/buttons_component.rb

```diff
@@ -36,7 +36,7 @@ def show_version_draft_cancel?
     end
 
     def show_first_draft_cancel?
-      work_version.first_draft?
+      work_version.deleteable?
```

Comment:
this method shows the discard draft button on the work edit page and needs to have additional logic now to deal with rejected items